source/v5/mqtt5_to_mqtt3_adapter.c

```diff
@@ -2226,23 +2234,22 @@ static void s_aws_mqtt5_to_mqtt3_adapter_subscribe_completion_fn(
             "id=%p: mqtt3-to-5-adapter, completing multi-topic subscribe",
             (void *)adapter);
 
+        AWS_VARIABLE_LENGTH_ARRAY(struct aws_mqtt_topic_subscription, multi_sub_subscription_buf, reason_code_count);
         AWS_VARIABLE_LENGTH_ARRAY(
-            struct aws_mqtt_topic_subscription, multi_sub_subscription_buf, suback->reason_code_count);
-        AWS_VARIABLE_LENGTH_ARRAY(
-            struct aws_mqtt_topic_subscription *, multi_sub_subscription_ptr_buf, suback->reason_code_count);
+            struct aws_mqtt_topic_subscription *, multi_sub_subscription_ptr_buf, reason_code_count);
         struct aws_mqtt_topic_subscription *subscription_ptr =
             (struct aws_mqtt_topic_subscription *)multi_sub_subscription_buf;
 
         struct aws_array_list multi_sub_list;
         aws_array_list_init_static(
             &multi_sub_list,
             multi_sub_subscription_ptr_buf,
-            suback->reason_code_count,
+            reason_code_count,
             sizeof(struct aws_mqtt_topic_subscription *));
 
         size_t subscription_count = aws_array_list_length(&subscribe_op->subscriptions);
 
-        for (size_t i = 0; i < suback->reason_code_count; ++i) {
+        for (size_t i = 0; i < reason_code_count; ++i) {
             struct aws_mqtt_topic_subscription *subscription = subscription_ptr + i;
             AWS_ZERO_STRUCT(*subscription);
 
```

Comment:
Check `reason_code_count` or `suback == NULL` so we don't try to access `suback->reason_codes` if it's NULL and set the `subscription->qos` to`AWS_MQTT_QOS_FAILURE`